include/picongpu/particles/shapes/NGP.hpp

```diff
@@ -17,36 +17,45 @@
  * If not, see <http://www.gnu.org/licenses/>.
  */
 
-
 #pragma once
 
 #include "picongpu/simulation_defines.hpp"
 
+#include <cstdint>
+
+
 namespace picongpu
 {
 namespace particles
 {
 namespace shapes
+{
+namespace detail
 {
 
-    namespace shared_NGP
+    struct NGP
```

Comment:
Why is `NGP` and `Counter` separate? Are they not both NGP?

Reply:
This I do not really know, and did not touch in thie PR. As far as I understood while working on this PR and writing comments to shapes, `NGP` is a "proper" (albeit a very bad for 2020) particle shape to be used in a PIC simulation. While `Counter` is a special one, only to be used for particle counting (instead of the actual shape) and with `ChargeAssignment` implemented specifically for this purpose. However, we already use it in the SAXS test example as the "main" shape as well, by mistake I guess. @psychocoderHPC is my understanding correct?